### ISSUE TITLE
Visual Studio modifies solution files created from the dotnet CLI.

### ISSUE THREAD

**Issue by g0t4** [dotnet/cli#6221]
## Steps to reproduce
```
mkdir test
cd test
dotnet new sln
dotnet new console -o console
dotnet sln add .\console\console.csproj
# commit to git repo before opening in VS
```
open the solution with Visual Studio (version info below), open diff of test.sln: 
![image](https://cloud.githubusercontent.com/assets/180189/24581925/eba47238-16f2-11e7-8ec6-bb36946911cc.png)

## Expected  behavior
I would assume Visual Studio would leave the file alone. Or, the template from `dotnet new sln` should conform with what Visual Studio expects. 

## Actual behavior
Visual Studio makes many changes, see above. 

## Environment data
`dotnet --info` output:

.NET Command Line Tools (1.0.1)

Product Information:
 Version:            1.0.1
 Commit SHA-1 hash:  005db40cd1

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.1

## System info:
![image](https://cloud.githubusercontent.com/assets/180189/24581904/8f2b5378-16f2-11e7-8530-64996a09729d.png)

## Visual Studio info:
Microsoft Visual Studio Community 2017
Version 15.0.26228.12 D15RTWSVC
Microsoft .NET Framework
Version 4.6.01586

**Comment by livarcocc** [comment]
@srivatsn @333fred Any idea why VS is modifying the file? Should we change ours to conform with what VS is changing it to?

**Comment by srivatsn** [comment]
The first change is odd - the project guid is being changed to the CPS guid - I thought that only happens when a project is added to the solution. We should look at that as a bug on the project system.

The second change is that VS thinks that the project builds for AnyCPU and that's true by default. Why does the CLI add the x64 configuration as the default?

**Comment by srivatsn** [comment]
Oh and the change to the sort order - I assume VS sorts sections a certain way. CLI should just generate it in that order.

**Comment by draganjovanovic1** [comment]
Hi, I have strange issue similar to this. I am writing F# on dotnet core on macOS. When I create console app and add it to solution using CLI, project is added with F2A71F9B-5D33-465A-A702-920D77279786 GUID. The same happens if I try to add project to the solution using VS for mac. With both approaches build works as expected.

However, when solution is opened on Windows 10 using VS 15.3, project GUIDs are being changed to 6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705 and everything works on Windows machine. When I move back to Mac and try to open the same solution it won't compile until GUID is reverted to F2A71F9B-5D33-465A-A702-920D77279786. Alternatively it will compile if I keep 6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705 project GUID but then I have to change backslashes "\" with slashes "/" in the project path!!!! 🤓 

Since we are using different platforms (Mac and Windows) to work on the same solution this issue causes a lot of pain.

**Comment by peterhuene** [comment]
The reason why VS thinks the solution file is "dirty" is that the `x86` and `x64` platforms do not exist for the project, but exist for the solution.  When the solution loads, Visual Studio has to do its best to map the solution platforms to an existing project platform, selecting the default "Any CPU" platform.

If the solution file is edited to remove the `x86` and `x64` solution platforms prior to opening, Visual Studio does not modify the solution file.  The CLI should not be creating these solution platforms by default.

**Comment by peterhuene** [comment]
I think I will fix `dotnet sln add` to map solution configurations/platforms that don't have a corresponding project configuration/platform to the first configuration/platform supported by the project (this is what VS does) regardless of whether or not `dotnet new sln` should initially contain the `x86` and `x64` platforms.

**Comment by ikourfaln** [comment]
Hi
@peterhuene, I see that this issue is resolved but Why Visual Studio 15.6.7 still modify solution file generated by dotnet CLI !!

![vs-solution](https://user-images.githubusercontent.com/9744226/39412013-0d85a3a4-4c0d-11e8-85bc-25401f29cd23.png)

Using dotnet CLI, I generate two projects and one solution file, then add these projects to the sln. All is fine when I use Visual Studio Code.
But when other contributors open the project using Visual Studio 15.6.7, the sln file changed, and that's affect ou versioning system because we don't want to track the sln file too.

**Comment by peterhuene** [comment]
Hi @ikourfaln,

The fix will be released with the upcoming `2.1.300` version of the .NET Core SDK; it should be present in the [Preview 2 release of the 2.1.300 SDK](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300-preview2).  As it is not yet shipping in a final release version of the SDK, Visual Studio updates do not, as of yet, contain this fix.

**Comment by ikourfaln** [comment]
@peterhuene 
Thank you for the reply, I'm sorry, I didn't pay attention to the milestone of this issue.
